src/tribler/gui/network/request/request.py

```diff
@@ -0,0 +1,126 @@
+from __future__ import annotations
+
+import json
+import logging
+from time import time
+from typing import Callable, Dict, Optional, TYPE_CHECKING, Union
+
+from PyQt5.QtCore import QObject, pyqtSignal
+from PyQt5.QtNetwork import QNetworkReply, QNetworkRequest
+
+from tribler.gui.utilities import connect
+
+if TYPE_CHECKING:
+    from tribler.gui.network.request_manager import RequestManager
+
+
+class Request(QObject):
+    GET = 'GET'
+    POST = 'POST'
+    PUT = 'PUT'
+    PATCH = 'PATCH'
+    DELETE = 'DELETE'
+
+    # This signal is called if we receive some real reply from the request
+    # and if the user defined a callback to call on the received data.
+    # We implement the callback as a signal call and not as a direct callback
+    # because we want the request object be deleted independent of what happens
+    # during the callback call.
+    on_finished_signal = pyqtSignal(object)
+    on_cancel_signal = pyqtSignal()
```

Comment:
`on_cancel_signal` is obsolete and can be removed without any harm. It was used originally to re-send the shutdown request in case it was canceled. But the current logic avoids canceling the shutdown request, so it is unnecessary to re-send it now.